scopesim/effects/ter_curves_utils.py

```diff
@@ -353,25 +353,32 @@ def scale_spectrum(spectrum, filter_name, amplitude):
     return spectrum
 
 
-def apply_throughput_to_cube(cube, thru):
+def apply_throughput_to_cube(
+    cube: fits.ImageHDU,
+    thru: SpectralElement | SourceSpectrum,
+    wave_cube: u.Quantity,
+) -> fits.ImageHDU:
     """
     Apply throughput curve to a spectroscopic cube.
 
     Parameters
     ----------
-    cube : ImageHDU
-         Three-dimensional image, dimension 0 (in python convention) is the
-         spectral dimension. WCS is required.
-    thru : synphot.SpectralElement, synphot.SourceSpectrum
+    cube : fits.ImageHDU
+        Three-dimensional image, dimension 0 (in python convention) is the
+        spectral dimension.
+    thru : SpectralElement | SourceSpectrum
+        Throughput curve, spectrum or filter.
+    wave_cube : u.Quantity["length"]
+        Wavelength axis of the cube.
 
     Returns
     -------
-    cube : ImageHDU, header unchanged, data multiplied with
-         wavelength-dependent throughput.
+    cube : fits.ImageHDU
+        Header unchanged, data multiplied with wavelength-dependent throughput.
+
     """
-    wcs = WCS(cube.header).spectral
-    wave_cube = wcs.all_pix2world(np.arange(cube.data.shape[0]), 0)[0]
```

Comment:
This function (`apply_throughput_to_cube`) is only called in two places: `TERCurve.apply_to()` and `SpectralEfficiency.apply_to()`. In the first case, we have the `CubeSourceField` object, which has a `.wave` property that does this. Although I just realized that in `TERCurve.apply_to()`, the `wave_cube` argument is not passed to `apply_throughput_to_cube` (changed it at some point and probably forgot to adapt there). This should have raises an error somewhere, but as I just saw in the Codecov report, that case isn't covered in testing. So I'll go back and fix this. In the case of `SpectralEfficiency.apply_to()`, I basically put the WCS-wave stuff there.

In short(ish), I added `CubeSourceField.wave`, because this is needed in a few other places (somewhere in a notebook I believe), and is generally useful to have as a shorthand. Making it redundant in `apply_throughput_to_cube` itself. The `SpectralEfficiency` applies to the FOV not the Source, so we have no `.wave` property there (perhaps we should?), so still need the WCS.